### ISSUE TITLE
Units becoming unavailble

### ISSUE THREAD

**Issue by Pel1can111** [jeatheak/Mitsubishi-WF-RAC-Integration#106]
I am currently unsure if this is a problem with my setup or a bug in HA or this plugin but I set up an automation to alert me to any changes in the error code reported by the indoor units. I'm not sure if it started happening after the latest update with this plugin or the latest version of HA but I am randomly getting notifications due to the units becoming unavailable. Looking at the logs they seem to just briefly become unavailable then return to normal. This is something that would not be noticable if it were not for the notifications I had set up.

Please could other users just check their logbook and see if their units are doing the same so I know if this is a problem with HA or something with my units?

Thanks

**Comment by drawgas** [comment]
I can confirm, after latest update - units started frequently becomming unavailable. Restarting HA not always helps

**Comment by jeatheak** [comment]
Hi,

This is indeed because of the latest update. There were some feature requests about the availability of the device when you power it off.

I implemented this change. But this also means that if a call from the home assistant fails the AC unit, it becomes unavailable. That a call fails is a known issue and cannot be fixed at the moment unfortunately.

So you can change your automation to only report after X tries or I will implement a disable or X count before the unit becomes unavailable. I'm not sure yet.

Sorry for the inconvenience but I think the ability to make the unit unavailable after some time should be there.

**Comment by Pel1can111** [comment]
Thanks for the update. Aside from the notifications its not been causing any trouble operating the units so I will modify the automation to prevent false notifications, still nice to know the cause.

Thanks :)

**Comment by drawgas** [comment]
Thanks for the update.
Yes, it would be nice to have the option "unavailable after x retries" as currently it's a bit too frequent :)

**Comment by 290989** [comment]
I'm having this problem quite often at the moment. Sometimes the device remains unavailable. I've tried turning off the fuse, but that often didn't help either.

**Comment by 290989** [comment]
An indoor unit was not available again for two hours today. I then deleted it and wanted to add it again.
I then got the following error:

![IMG_0355](https://github.com/user-attachments/assets/855a6da3-92d4-48b1-bb31-8818fcebc971)

After I turned the fuse off and on again, it worked again. Does this help to solve the problem?

**Comment by jeatheak** [comment]
Hi, 

As far as I can see this is not a problem I can solve. The unit is really unavailable at that moment. If max retries have been exceeded it means that it cannot connect to the unit (get no response).

Did you try the smart-m air app at the same time? 

The only thing I can try to add is an X count before putting it unavailable. 
But I think the unit is still unavailable and won't respond to your commands.

Before the update this was also happening but it was never visible in homeassistant. 

Do you have more instances (homeassistant, apps) polling the unit? 
This can also cause some connectivity issues. The wifi module really does not like multiple connections as far as I know.

**Comment by 290989** [comment]
> Edit: I maybe can add an option to disable the availability option.


That would be nice. Thank you 👌🏻

**Comment by 290989** [comment]
> > Edit: I maybe can add an option to disable the availability option.
> 
That would be nice. Thank you 👌🏻

I have only the Smart M-Air App & Home Assistant. 
Could it be, that the many pings if it’s available is to much & the unit is unavailable then?

**Comment by Rorrik404** [comment]
> As far as I can see this is not a problem I can solve. The unit is really unavailable at that moment. If max retries have been exceeded it means that it cannot connect to the unit (get no response).

> Edit: I maybe can add an option to disable the availability option.

Agreed the unit really is offline at those points. I've tested mine via a post request and get a failure to establish connection. I've found that putting a DNS block on iot.smartmair.com has decreased the timeouts. Ultimately I think it's badly coded firmware on the AC side.

Adding an option to disable the availability check would be very useful. Likewise maybe adding an extending delay (to a point) for rechecks. E.g. GetStatus -> Fail -> Try again -> Fail -> Wait 30 seconds -> Fail -> Wait 1 Minute... Cap at checking every 5 minutes until its back?

**Comment by 290989** [comment]
> Do you have more instances (homeassistant, apps) polling the unit? This can also cause some connectivity issues. The wifi module really does not like multiple connections as far as I know.
> 

Do you know what are the 2 Accounts?

![IMG_0453](https://github.com/user-attachments/assets/250ec6be-4c86-4bcd-9744-bd0fc58694ab)

**Comment by jeatheak** [comment]
> Do you know what are the 2 Accounts?

Yes this are the number of registered devices (apps) with the airco unit.
As far as I now you can have a maximum of 4 deviecs registeres (so for example: 3 smart-m air apps (different phones) and 1 homeassistant connection)

If it has reach 4 than you need to remove one.

> could this be multiple connections?

In theory yes but only when the smart-m air app is activly open on the phone. It even sometimes gives a message that another instance is connected. But this will be resolved after some time. So this should not be a problem.

**Comment by jeatheak** [comment]
> Agreed the unit really is offline at those points. I've tested mine via a post request and get a failure to establish connection. I've found that putting a DNS block on iot.smartmair.com has decreased the timeouts. Ultimately I think it's badly coded firmware on the AC side.

Yeah indeed I think it is a weirdly designed firmware. I have blocked the complete internet access of the AC's and still have it somethimes.

> Adding an option to disable the availability check would be very useful. Likewise maybe adding an extending delay (to a point) for rechecks. E.g. GetStatus -> Fail -> Try again -> Fail -> Wait 30 seconds -> Fail -> Wait 1 Minute... Cap at checking every 5 minutes until its back?

This is a good suggestion. I will look into it.

**Comment by Rorrik404** [comment]
@jeatheak 
Adding an extending delay getAirconStat check would likely solve 95% of the issues of the device becoming unavailable in HA. There's nothing you can do about it when the device is actually playing up (as discussed above). 

**That said a possibly "quick and dirty" fix would be add a "Force Recheck" button to the device page that calls the async_update function.** 

I found when testing that the device can become available again (constant posts going through without timing out) however HA doesn't update for quite a while. The only quick solution to get it available again is to restart HA so it does a getAirconStat when HA comes back up. Adding a "Force Recheck" button wouldn't solve the issue of it going offline, but would provide a very quick way to run an "has it started behaving again" check and avoid restarting HA.

**Comment by jeatheak** [comment]
It takes a bit more time and testing. 
Sorry for the delay. Have a very busy period at the moment.
You can follow the progress at branch `2024.7`

**Comment by rogermeijer** [comment]
I think I am facing similar issues with one unit in particular that becomes unavailable and stays like that. While waiting on 2024.7 (although I am already running 2024.7-beta.2), is there a way to get the unit available again? Restarting HA, redownloading integration, disabling/enabling device, nothing works. Any tips? Thanks a lot!

**Comment by Pel1can111** [comment]
> I think I am facing similar issues with one unit in particular that becomes unavailable and stays like that. While waiting on 2024.7 (although I am already running 2024.7-beta.2), is there a way to get the unit available again? Restarting HA, redownloading integration, disabling/enabling device, nothing works. Any tips? Thanks a lot!

This sounds like an actual problem with the unit, perhaps poor wifi signal in that area? I would suggest trying to access the unit through the smart m-air app and see if its still available there when not available in HA. If its not working via the app then it points towards an issue with the unit and not HA.

**Comment by 2112b** [comment]
Roger, if you mean by unavailable that the unit does no longer responds in
HA...
This happend here after operating the unit with the remote.
(the tempgauge changed colour to green I think)

Eventually, switching the unit off with the Remote returned the control in
HA (after a short while)

**Comment by rogermeijer** [comment]
Thanks for the quick responses, your comments made me look (again) into the given IP addresses and for some reason the IP address of this specific device changed. After reconfiguring the IP address and restarting HA it is available again 🫣

The previous time it went "unavailable", while still working in the Smart M-Air app, it was not an IP issue. It did go back online after redownloading the integrations.

The second time it stayed unavailable, so I skipped the IP check. Sorry for the confusion! Now I need to find a way to fix the IP-address to the device with my current set-up 🧐

**Comment by perplexityjeff** [comment]
Hi,

I have noticed it mostly with 1 unit that in HA it is seen as 'unavailable' however using Smart M its perfectly controllable which seems weird. Also I see no drop in wifi connections. I just installed the integration but it seems that the checks are a bit to aggresive for availability?

**Comment by rfx77** [comment]
Same problem here.

The Unit is working in the M-Air App without any issues but it refuses the local connection. I tried with curl

```
curl http://172.27.102.103:51443
curl: (7) Failed to connect to 172.27.102.103 port 51443: Connection refused
```

Th IP is correct
The problem seems to be that if you do too many connection to the unit via port 51433 it stops responding for some time.
After a while it starts responding again.

Maybe your aviability check is to agressive. Is it possible to stop the availability check or to set the rate?

**Comment by dom404** [comment]
I see this on all 7 of my units 

1 or 2 will stop responding in HA; however, the Smart M-Air App always works even when the port appears to be down.

**Comment by neworld** [comment]
Sorry if this is the wrong thread to continue. I recently encountered the same problem. The unit had been working perfectly for 1.5 years with no single failure. However, it suddenly became unavailable in HA, and restarting HA and A/C did not help.

I banged my head against the wall for two days until I tried to launch automation, which I had disabled a few days ago, and bam, it works again. 

So, status updates work only if the same device requests a set state. 

It is not enough to set the state via mobile. I am considering making an automation to set the state each day. It would be nice to find a way to set same state

**Comment by dom404** [comment]
> Sorry if this is the wrong thread to continue. I recently encountered the same problem. The unit had been working perfectly for 1.5 years with no single failure. However, it suddenly became unavailable in HA, and restarting HA and A/C did not help.
> 
> I banged my head against the wall for two days until I tried to launch automation, which I had disabled a few days ago, and bam, it works again.
> 
> So, status updates work only if the same device requests a set state.
> 
> It is not enough to set the state via mobile. I am considering making an automation to set the state each day. It would be nice to find a way to set same state

How can you set the state if the device is unresponsive in HA? Does HA not complain the device is off?

**Comment by neworld** [comment]
> How can you set the state if the device is unresponsive in HA? Does HA not complain the device is off?

I have automation to enable the air conditioner based on time each day, but I disable it during cold weather because I don't need it. 

I will do extra tests on this

**Comment by Pel1can111** [comment]
I have been running the latest beta for a few days now with notifications re-enabled and I’m yet to get one error so from my standpoint this issue can be closed. Thanks

**Comment by rfx77** [comment]
I also had this issues but i could solve them. As i said above the units become unavailable sometimes when the do a dhcp renew. 

I did the following things:
1) disabled cloud connection in the Mitsubishi app and switched to local connect
2) denied any internet access for the HVAC devices
3) kicked the devices out of wifi every 30min (automated with my ubiquiti)

I think 1 and 2 did the trick. 3 seems to be optional 

Greetings,
Franz

**Comment by dom404** [comment]
> * disabled cloud connection in the Mitsubishi app and switched to local connect
> * denied any internet access for the HVAC devices

I suspect most people are using the Mitsubishi's that don't have a Cloud connection, so no internet requirement.

Booting the device off the wifi will make no difference.

Why don't you fix the IP of the device? seems like (in your case) the easier option?

**Comment by rfx77** [comment]
The devices have static dhcp assigned but they do renew anyway every hour no matter how long you set the lease. So this is in their Firmware as it seems.
If you confgure the WF-RAC for local access (App) they are connecting to the internet also. I had to block them.

**Comment by dom404** [comment]
> The devices have static dhcp assigned but they do renew anyway every hour no matter how long you set the lease. So this is in their Firmware as it seems. If you confgure the WF-RAC for local access (App) they are connecting to the internet also. I had to block them.

Different devices connect and perform in different ways.

M-Air is the app quite a few will use - No internet access or cloud function/access. Yours do connect, but many do not. Mine certainly do not renew their lease every hour.

**Comment by rfx77** [comment]
> > The devices have static dhcp assigned but they do renew anyway every hour no matter how long you set the lease. So this is in their Firmware as it seems. If you confgure the WF-RAC for local access (App) they are connecting to the internet also. I had to block them.
> 
> Different devices connect and perform in different ways.
> 
> M-Air is the app quite a few will use - No internet access or cloud function/access. Yours do connect, but many do not. Mine certainly do not renew their lease every hour.

Do you have connectivity problems now or did you solve them?

**Comment by dom404** [comment]
Trialling the latest beta. To soon to say

**Comment by rfx77** [comment]
Maybe you could check on your Firewall if the HVAC units really dont try to connect to the internet. In my situation they always did and when i denied them the problems went away.

**Comment by dom404** [comment]
> Maybe you could check on your Firewall if the HVAC units really dont try to connect to the internet. In my situation they always did and when i denied them the problems went away.

Please stop making assumptions or asking questions that have already been answered

**Comment by dom404** [comment]
I can see that even with 2024.7 I have got one of my units being unavailable at the moment.

51443/tcp closed unknown

M-Air app works as expected

**Comment by jeatheak** [comment]
Please don't create a new issue for the same problem. @dom404 .

It's getting a bit frustrating for me at the moment because I have 2 units at home and they never disconnected in 2 weeks or even more. So it is really difficult to debug/fix this.

You sure it is not your network? The big difference between the smart m air app and home assistant is, that the app only connects when you open the airco in the app. And home assistant polls (all) they airco constantly. 

I even put 1 airco on a remote socket I can turn on/off. it now always reconnects nicely, even after being offline for 1 day.

So to conclude: I will try to look into it (again) but it will take time and for now I don't know if I can find a solution for you.  So sorry for that.

(My airco's are blocked from internet access and are in an isolated 2.4ghz network)

**Comment by dom404** [comment]
I am happy to do any testing you require/need. But it's catching the units when they are offline/unavailable. I usually only see if HA reports that it was unavailable in the past while looking at the log book.

I have 7 units used for main house heating so I might notice more than most. 

I will increase the retry limit from 3 - any recommendations 

I can see it was reported as unavailable for 20 minutes during this time it was available and responding to M-Air (I believe there is a 60-second timeout between devices talking to the units) 

I don't see my units talk to the internet at all. I did notice that there is an option for remote access which I don't use (Set to local) so this might be why.

**Comment by Pel1can111** [comment]
I really struggle to see there are any issues with the plugin at this point causing this. Prior to the update (with the aircon's on my list of devices I get notified if they go offline) I was getting between 2-6 notifications every day that the aircon units were going offline.

Since I installed the update (little over 2 weeks I think) I have received 1 notification and that was caused by me turning the breaker off for that circuit to install a new plug. I have 3 internal units so its a pretty good test and I have the retry limit set to 3 so I have no doubt the units are not responding a bit like before but this is gettings ironed out by the retries.

I just looked and my app is set to remote access (not that I use it). I have not bothered to block their internet access, just have them on a seperate vlan other devices.


Thing I would suggest you try.

Turn the power off to the airon, leave it off a few minutes and turn it back on and see if that helps. I have in the past seen a unit stop responding for a long period of time and this was the easiest fix.

If that does not help, I would try and place a wifi access point next to the unit and see if that rules out a signal issue.

**Comment by dom404** [comment]
My units are a multi. I would need to turn the main breaker off outside to all the units to turn off one.

I cannot rule out a wireless issue and it was the first thing I tested. However, the M-Air App always works over the wireless. even when the port is showing as closed. Assuming that the M-Air App and the Plugin are using the same port for communication. 

I thought it was the service stopping/crashing but the App rules that out.

**Comment by Pel1can111** [comment]
might also be worth checking they are on the latest firmware. I think 131 is the latest

**Comment by dom404** [comment]
I have just checked.

All units are running 131 - In 24 hours this is the only reported unavailable time across all 7 units.

My units are currently set up for winter. They will turn on for an hour if the temperature of the room drops below x plus a schedule for when people are home. I will watch and see if this pops up again.

I should note that this is not a common trait and Is sporadically annoying when caught. I am tempted to proxy the Mair app and see why it works, but nothing else does.

**Comment by dom404** [comment]
So, I have been logging the devices on the wifi.

I can see that the AirCon Units disconnect from the wifi once an hour every hour, they are the only devices that do. They instantly reconnect and then are good to go another hour. It's very strange. They also are not all at the same time 

i.e. one unit will disconnect and reconnect at 12:30, 13:30, 14:30, etc another will be at 12:16, 13:16 etc all 7 units do this.

I don't have any evidence to confirm this is a contributing factor for the unavailable status. But I will continue to monitor.

**Comment by Pel1can111** [comment]
I'm almost certain this is the reason for the errors. Before the updated version if the integration tried to talk to the AC while it was reconnecting to WiFi it would time out and fail, the new version with retries "solves" this.

The issue with the hourly disconnects appears to be a issue with the firmware/hardware and I would suggest everyone bugs the manufacturer to solve this.

Other users with the issue are moaning about it here as well - https://community.home-assistant.io/t/mitsubishi-wifi-module-wf-rac-smart-m-air/411025/192

The firmware update a while back (131) seems to have been released to solve this and some users report it did for a while but the issue returned. All my units reconnect every hour looking at my unifi dashboard.

**Comment by dom404** [comment]
I am sure you are correct but I cannot get the times to match. I can see that HA reported the device as unavailable at 9:53:56 AM, but the reconnect time was 9:44 AM, 10:44 AM etc.

It's possible what I am now seeing is the delay between the two (I have the retry set to 10) 

What I need to try is scanning the port. When the plugin reports the unit as Unavailable the port is always closed. It is like the service is restarting. I don't know. Just because the unit is connected, the port is up. 

At least I know when the device is going to reconnect.

I still need to check what the app is doing so it is able to work but nothing else can.

**Comment by rfx77** [comment]
As i said above i had the same issues and the same behavour and this steps solved it 100%

https://github.com/jeatheak/Mitsubishi-WF-RAC-Integration/issues/106#issuecomment-2393023520

I have a new multisplit with 4 indoor units and wifi built in. i am on the newest firmware but this did not solve the issues.

i cannot speak for other unit types but mine always tried to connect to the mitsubishi cloud server no matter if they where in local or cloud mode. after making sure they are not able to do this anymore the problems went away. 

i also monitor the units with a recurring curl call to be sure they dont go away.

**Comment by dom404** [comment]
@rfx77 

I am similar except I have 7 units over 2 Multi Splits - What I am seeing is at most, 1 unit dropping per day for 20-60 minutes (according to HA) 

I think I have two questions that I need to answer. How does the App work when the plugin cannot? When does the port close? 

Mine don't talk to the internet - But that's more my setup, if yours does, or try, even in local mode.

I would also like to say that this is not causing me an issue - I have not had a unit (since increasing the retry) refuse to switch on or off when it was supposed to according to the schedule I set. This is more a case that it could cause an issue and I would like to resolve it fully.

**Comment by scooter306** [comment]
Have the Same issue with 2 devices. Blocking internet access via UniFi won't help, in local mode they are even worse available, so switched back to remote. 
As long as I don't add them to HA via this addon they are reachable in Smartair-App. The trouble starts when I activate it in Homeassistant. 

What I have noticed so far while resetting WiFi often to get rid of the problem: devices seem to have internal protection of getting polled to often. 
Maybe you could create a beta with an option to poll data just every 10 minutes or make it selectable how often data should be polled?

Maybe the device has something like ddos-protection with newer firmware and closes connection every X polls within y minutes. Could be worth a try, if you say it's not too much work to implement a poll limiter/timer?

Thanks in advance. 
I appreciate your work, love the plugin, used it last winter to keep the house warm and in the summer to keep it cool. 
Now it's a hard to get a reliable warming as the devices are always not reachable and has to be switched on and off by remote. 

- [ ] Next workaround for me would be to buy IR-Devices to get a reliable heating again, if it is allowed to use IR and reliable in same sentence 😂
At least I could resend the on-command via IR when not within 5 minutes the temperature raises in the room 😁

Thanks for your hard work!

**Comment by rfx77** [comment]
Hi!

I am executing this command on all of my 4 devices: `curl -s http://[IP of device]:51443` every second so that i can check te availability. When i get `Not supported this command` than everything is fine. I had no issues with stability when i execute this many requests. I also changed the code of this integration to test various intervals but had no success. 

Maybe you can execute this command periodically for some days to get a picture of how often the devices become unavailable and for how long they stay unavailabe. my devices stayed unconnected until they reconnected to the wifi after one hour.

**Comment by rfx77** [comment]
@dom404

before i had my solution where i kick the devices periodically they reconnected almost exactly every hour. i watched the process and it didnt matter if they where cloud or locally connected.

Every hour:
1) Network drops and Devices tries DHCP-Renew and Wifi reconnect (sometimes in unter 1s)
2) Network tries to reach some cloud-addresses
3) Device is available at port 51433 (or not)
4) 60 min wait
5) restart at 1

If you dont have connectivity aufter point 3 you will not get it back for 60min normally.

In my case i am not sure if the local connection with the m-air app ist working when the devices are not available on port 51443 as the tcp-port is closed. Cloud always works if i enable it and if i allow them internet connection. It is also not a timeout issue. i tried a lot of settings with my curl monitoring. if the are not reachable that stays that way until they reconnect themselve or are forced to do so.

There seems to be a big bug in the FW of those devices.

**Comment by scooter306** [comment]
For me the devices are always available in the app now since I deactivated the HA integration. As soon as I restart it I get the ACs to be disconnected soon and be unreliable. 
Maybe there are different bugs, concerning different kind of AC generations?

**Comment by dom404** [comment]
@rfx77 

I have played with various settings and configurations over the last week or so. 

In local mode in the M-Air App
I see only 1 device out of 7 has any drop for 1 hour in any 36-hour period reported in HA. Usually the same unit
I see them reach out to the internet at https://iot.smartmair.com/ every hour so I allow-listed the address from my filtering and allowed them access to the internet to that domain.

I see, as you:

client disconnects and reconnects and reaches out to https://iot.smartmair.com/every hour


The times for the client disconnect and reconnect do not coincide with the times that HA has them as being off-line

They will always come back when they next disconnect/reconnect. This is typically 50 minutes.

I am at one unit down, in HA for 50 minutes every other day. 

I do believe this is an issue in the device and not an environmental issue. I can, however, make it an environmental issue by impacting the access to the URL iot.smartmair.com or allowing roaming of the device on the wifi.

This is rare enough not to be an issue, although I can see that it could be as it starts to get colder.

I am going to test remote access for a week and see if there is any difference in the data. After that, I will look at what the Smart M app is doing to bypass the issue.

**Comment by Dream1975** [comment]
I have 3 airco units (1x srk50zs-w and 2x srk25zs-w) for several years now and the connection has all that time been flippering always. See below the log I just pulled which is only 4 minutes (but this goes on 24/7).

![Screenshot 2024-10-29 at 22-15-58 Unifi UCG Ultra - UniFi Network](https://github.com/user-attachments/assets/dbdad2cd-38ea-49e9-ac03-05c81679c4f6)

When the airco's were installed I had a Netgear (WiFi) router + used only the smart-m air app and the flippering was already there. The past year I went from netgear to Unifi (with multiple Access points)  and from the smart-m air app to home assistant with this great integration, but the flippering never changed (even though 2 airco's are right next to WiFi access points). So except from the WF-RAC module all other variables have changed.

In further have in total 47 devices connected to the access points (probably around 40 of which IoT goodies) and none of them show up as losing connection in the logs (thus have stable connections, even some of them being at the utter edge of the WiFi signal). Except for the 3 airco's that are real log fillers disconnecting/connecting every minute or so all day long.

So, unless there is a setting for the Airco/WF-RAC I don't know about and is configured wrong... I have to come to the conclusion that it has nothing to do with the integration but, in fact, has to do with the WF-RAC module's themselves not really being stable in having/holding a WiFi connection.

**Comment by dom404** [comment]
@Dream1975 

you have some WiFi issues looking at that small output. Certainly concerning enough to investigate.

**Comment by Dream1975** [comment]
> @Dream1975
> 
> you have some WiFi/network issues looking at that small output. Certainly concerning enough to investigate.

I just got around to look a little further in the fact my airco's (dis)connect every minute....

Airco wise there is just not that much that can be wrong as the settings are minimal. I logged into the accespoint and did a tail -f /var/log/messages (I just noticed all three airco's connect to the same AP even though for one of the airco's there is another one nearby).

It's all a bit jibberish for me, would someone mind to have a look if a point of interest rises out of this 4 minute log regarding the connection?
[Log airco.txt](https://github.com/user-attachments/files/17589055/Log.airco.txt)

Ps. Hope you don't mind asking a small beside question in the thread?

**Comment by dom404** [comment]
@Dream1975 

just had a Quick Look since this is not the place for this one

recommend limiting your dB-80+ is to high. No idea what the roaming threshold is on these units but I suspect your past it which would exasperate your issues.

**Comment by alexnikgr** [comment]
Hello all,

It happened to me as well. It seems that the device has an inbuilt primitive anti-DoS mechanism to ensure that it does not get out of service by a bad integration or something and it is getting triggered by the HA integration.

Could it be that the status requests are very often, or the requests are one too many? Otherwise the device is connected, alive, answering to ping and responding to cloud actions.

**Comment by scooter306** [comment]
Same for me

**Comment by mmaarrkk02** [comment]
@Dream1975 
Based on your logs, here are some possible reasons for the Airco's wireless network instability:

1. RSSI (Signal Strength) Issues
You observed that the RSSI (average signal strength) for multiple STAs (such as Airco Office, Airco Livingroom, and Airco Bedroom) is between -60 and -85 dBm.

-60 dBm is fairly stable, but -85 dBm is near the unusable range, which may cause frequent disconnections.
Solution:

Check the placement of your Access Points (APs). Try to reduce obstructions or interference, or add additional APs to improve coverage.


2. DHCP Allocation Overlap
The logs show that devices frequently receive IP addresses in the 192.168.10.1 and 192.168.2.x ranges, which may indicate a DHCP misconfiguration or multiple DHCP servers operating in the same network.

Problem: If there are multiple DHCP servers, devices might be assigned incorrect IPs or frequently change IPs.
Solution:

Check the DHCP range configuration to ensure only one DHCP server is responsible for assigning IPs within that subnet.

**Comment by Dream1975** [comment]
Hello mmaarrkk02,

Thank you so much for investing your time in analyzing the problem. In the meantime, I factory reset all 3 WF-RAC modules and the Smart M Air app and install them from scratch. This has led to 2 of the 3 now not having the issue anymore (only the hourly reconnect). 

The 3rd one keeps reconnecting every minute or so. This one is within 4 meter of the access point, so it shouldn't be the reception. Also, not the AP itself as one of the other 2 is also connected to this UAP AC-LR (with much greater distance even). Since then I factory resetted this module several times again and connect with all variables I can think off (also to the other AP etc), but this one keeps doing it. As the other 2 are now much more stable I haven't tried factory resetting all again (in fear losing this stability again).

So there is some progress, but just can't figure out why the 3rd is still flippering.

**Comment by Dream1975** [comment]
I have one Unifi cloud gateway Ultra that also is the only DHCP server. Furthermore 3 8-ports switches throughout the house and 2 UAP AC-LR's (both connected through PoE from the switch).

Ans it remains strange first all three did it, and now only one still. And as the flippering one is connected to the same AP as another without having the problem anymore it can't really be AP related (which I thought at first, thus trying several configs as far as possible without having to reset the 2 working ones).

Even switching UAP didn't make a diffreence. I did force the 2 working ones both on another UAP (as well as the flippering one) but it remained the same. The 2 ones not having it anymore working great no matter what UAP or something while the 3rd one keeps flippering. I would almost begin to think it is this WF-RAC module weren't it that previously all three did it.

**Comment by mmaarrkk02** [comment]
The logs you provided earlier seem to indicate a network issue. The two devices that are functioning normally are likely farther from the AP, so they haven’t experienced AP switching. However, the problematic device happens to be able to connect to two APs, which caused the issue.

If you're willing to provide the Unifi configuration settings, I’d be happy to help you investigate the cause.

**Comment by Dream1975** [comment]
That would be great. I did lock all three to the most nearby AP. 
Airco1: In the livingroom (locked to AP also in the livingroom) = now working OK
Airco2: In the bedroom (locked to AP in attic) = now working OK -> Airco is between AP livingroom and attic distance/obstacle wise
Airco3: In the attic (also locked to AP in attic)

Accespoints: AP livingroom (Unifi UAP WK) and AP attic (Unifi UAP Z)
![1 Access points](https://github.com/user-attachments/assets/831fb75a-fd74-49cb-9394-c9995b68588c)

Airco's
- Attic (zolder) has the problem
- Bedroom (slaapkamer) is now working fine since the reconnect
- Livingroom (woonkamer)  is now working fine since the reconnect
![2 Airco's](https://github.com/user-attachments/assets/091b85bb-044e-41ac-aa3b-7abae435a317)

Wi-Fi settings:
![3 WiFi](https://github.com/user-attachments/assets/9c042be3-79fa-418a-b954-60c5a3025566)

Network settings:
![4 Network](https://github.com/user-attachments/assets/ffef35ac-4310-4aff-a4b5-c309b29b83e7)

**Comment by Dream1975** [comment]
Also decided to investigate myself a bit further, by setting op a syslog recipient. What I notice is the flippering one having a lot of
thhese messages.

DESCR="no rule description" IN=br2 OUT= MAC=################### SRC=192.168.2.193 DST=224.0.0.251 LEN=32 TOS=00 PREC=0x00 TTL=1 ID=8 PROTO=2 MARK=1b0000

and

DESCR="no rule description" IN=br2 OUT= MAC=################### SRC=192.168.2.193 DST=192.168.2.1 LEN=63 TOS=00 PREC=0x00 TTL=255 ID=4 PROTO=UDP SPT=8739 DPT=20201 LEN=43 MARK=1b0000

When looking at the orther 2 they als have them but far, far less (which makes me suspect this is the trigger maybe as it quite matches the frequency of reconnections). 

Would seem to be multicast related then. But what I don't understand is why the attic one multicasts so often while the other 2 do it enourmously less?

**Comment by mmaarrkk02** [comment]
From the configuration, it seems there shouldn’t be any issues.

Here are some steps you can try to resolve the problem:

Create a new Wi-Fi SSID specifically for Airco3 to use.
Use the Smart M-Air app to check if there’s a firmware update available.

**Comment by rfx77** [comment]
@Dream1975

I dont think you have network issues. depending on the model of you indoor units they disconnect regularly and sometimes dont come back. This seems to be by design but badly implemented

Here you see what the behaviour is on mine: https://github.com/jeatheak/Mitsubishi-WF-RAC-Integration/issues/106#issuecomment-2441255802

Since you have a unify device you could just use my stretegy. Deny them all internet access (double check this) and forcefully reconnect them every 15 minutes or so via scripting the unify-controller. This solved all my problems.

**Comment by Dream1975** [comment]
Reconfigured the flippering airco in another Wifi network as well as VLAN (and IP range) as suggested by @mmaarrkk02 this morning. And this one is now also connecting for an hour (and then reconnecting as do the other 2, but seems normal reading this topic). Ive been monitoring it all day.

It is strange as in the old Wifi/VLAN is removed and reconfigured this one dozens of times with always the same flippering result. The only thing I can think of at the moment as that having 3 MHI airco's in the same Wifi/VLAN is too crowded (with the multicast) or something leading to a disconnect.  But maybe some of you have more of a clue in what it could be.

I will, for now, keep monitoring if it remains this way (with one in one VLAN/WIFI and 2 in another). For home assistant is doesn't matter as long as I allow communication between the VLAN's. If I'm in a courageous mood after monitoring for a while I might try to reconnect the other 2 also in the new VLAN to see if it starts happening again.

@rfx77 If I deny them internet access the flippering starts again (on all three) and stops when they do have access again. Haven't looked further in killing the connection and forcefully reconnect because at how it is running now (one reconnect per device per hour) is a great improvement with 99% availability. The minutely reconnects were a real pain in the ass because they were almost never all three online (because of the reconnecting and starting up) and needed top give commands several times waiting a few seconds. This is now a thing of the past 😊

**Comment by mmaarrkk02** [comment]
I suspect that the flipped airco detects both the living room and attic Wi-Fi access points simultaneously and keeps switching between the two Wi-Fi APs.

You can try setting the new Wi-Fi network to the same VLAN as the existing Wi-Fi network.

I would recommend not using the same Wi-Fi SSID for IoT devices across multiple APs. Each AP should have its own IoT SSID, as IoT devices do not need to roam. It will be clearer which AP each device is connected to, making troubleshooting easier.

**Comment by Dream1975** [comment]
That could be the case. It is very strange now, mostly a reconnect after 15 minutes up to 45 now. But then it can happen sometimes it gets stressed and has several reconnects in a short time.

Enclosed a syslog I was running for this MAC address. It runs a while pretty OK, then a period of enclosed syslog (mpstly 10 minutes or so) and then for a while longer connects.
[all_2024-12-22-18_25_0.csv](https://github.com/user-attachments/files/18222731/all_2024-12-22-18_25_0.csv)

Ps. No idea why there are radius mentions, I don't use that.

> I suspect that the flipped airco detects both the living room and attic Wi-Fi access points simultaneously and keeps switching between the two Wi-Fi APs.

That's also an idea. Can't hurt to set up a temp Wifi network on one UAP and reconnect this one (if your idea is correct then it should stop). Not home for a few days so to be continued.... (it is very doable now, but always like to learn and puzzle maybe finding out what it is).

**Comment by Dream1975** [comment]
> I suspect that the flipped airco detects both the living room and attic Wi-Fi access points simultaneously and keeps switching between the two Wi-Fi APs.
> 
> That's also an idea. Can't hurt to set up a temp Wifi network on one UAP and reconnect this one (if your idea is correct then it should stop). Not home for a few days so to be continued.... (it is very doable now, but always like to learn and puzzle maybe finding out what it is).

Unfortunately not, It is now running on its own Wi-Fi which only one access point is broadcasting and is nearby, but still hiccups.

**Comment by Dream1975** [comment]
I have a firewall rule that drops all invalid packets from LAN to WAN and noticed below log entries from time to time. THe strange thing is only the one having hiccups are in there sending invalid packets to 54.76.72.158 (an Amazon AWS IP) the other 2 don't. I'm beginning to think this one might have another firmware version or something. That would explain a lot.

> DESCR="Drop invalid local packets" IN=br3 OUT=ppp0 MAC=9e:05:d6:6f:2b:52:ec:0b:ae:9f:4xxxxxx SRC=192.168.3.193 DST=54.76.72.158 LEN=40 TOS=00 PREC=0x00 TTL=254 ID=36 PROTO=TCP SPT=49154 DPT=8883 SEQ=12552 ACK=2013318675 WINDOW=2920 ACK RST URGP=0 MARK=1b0000

**Comment by mmaarrkk02** [comment]
You can try:
1. Allowing AirCo LAN to WAN.
2. Updating the firmware using the Smart M-Air app.

**Comment by alexnikgr** [comment]
I have a feeling that the above comments do not correlate with the issue.

I have tried the newest build and the issue is still appearing randomly. Firmware 010/131.
